src/Columns/ColumnFunction.cpp

```diff
@@ -93,14 +93,7 @@ void ColumnFunction::expand(const Filter & mask, bool inverted)
 
 ColumnPtr ColumnFunction::permute(const Permutation & perm, size_t limit) const
 {
-    if (limit == 0)
-        limit = size_;
-    else
-        limit = std::min(size_, limit);
-
-    if (perm.size() < limit)
-        throw Exception("Size of permutation (" + toString(perm.size()) + ") is less than required ("
-                        + toString(limit) + ")", ErrorCodes::SIZES_OF_COLUMNS_DOESNT_MATCH);
+    limit = getLimitForPermutation(*this, perm, limit);
```

Comment:
Do we need `getLimitForPermutation` here if we call `column.column->permute` that will do it again?

Reply:
Yes, it's used later in `ColumnFunction::create`.